tools/clang/test/CodeGenSPIRV/vk.combined-image-sampler.hlsl

```diff
@@ -25,6 +25,13 @@ Texture2D<float4> tex2 : register(t2);
 // CHECK: OpDecorate %sam3 Binding 0
 SamplerState sam3 : register(s0);
 
+// CHECK: OpDecorate %tex4 DescriptorSet 0
+// CHECK: OpDecorate %tex4 Binding 4
+[[vk::combinedImageSampler]]
+Texture2D tex4 : register(t4);
+[[vk::combinedImageSampler]]
+SamplerComparisonState sam4 : register(s4);
```

Comment:
The vulkan specs seems to allow using `sam4` with `tex2` for ex, but this crashes at the legalization step in the current state. Maybe that's why it was disabled?

Reply:
Good catch. That's an invalid use case because once a sampler is used in a combined image sampler, it can't be used with other textures at different descriptor bindings. That's because a SPIR-V sampler can't be extracted from the combined sampler. We should fail with something more helpful though, so I'll take a closer look.

Reply:
Wait, is that the case? My understanding of the Vulkan spec is the opposite:
(https://registry.khronos.org/vulkan/specs/1.2/html/chap14.html#descriptorsets-combinedimagesampler)

```
If the descriptor refers to a sampler that performs [Y′CBCR conversion](https://registry.khronos.org/vulkan/specs/1.2/html/chap13.html#samplers-YCbCr-conversion), the sampler must only be used to sample the image in the same descriptor. Otherwise, the sampler and image in this type of descriptor can be used freely with any other samplers and images.
```

Reply:
Oh, we just don't handle that error case at all. `tex0.SampleLevel(sam1, float2(1, 2), 10, 2)` also fails at legalization. I opened a new issue to track it: https://github.com/microsoft/DirectXShaderCompiler/issues/6346.

I don't think it needs to block this PR though since it's an existing bug that isn't affected by this change.

Reply:
Oh right, thanks for checking!

Reply:
Sorry, I didn't refresh the page to see your new comment before posting mine. I'll take a look at the spec later today, maybe we shouldn't have that restriction.

Reply:
Thanks, haha, I'm all out of sync on the threads today. I guess either way it's not blocking this PR, but we should investigate further... I'm not actually sure how we'd differentiate between whether or not the sampler performs Y′CBCR conversion, so I think it still might be the right call to disallow it in HLSL. But either way we should fail more gracefully than we do now.